src/hooks/useCourtContracts.js

```diff
@@ -410,54 +427,103 @@ export function useAppealFeeAllowance(owner) {
   const feeTokenContract = useFeeTokenContract()
 
   useEffect(() => {
+    let cancelled = false
+
     const getFeeAllowance = async () => {
       if (!feeTokenContract) return
 
       retryMax(() => feeTokenContract.allowance(owner, disputeManagerAddress))
         .then(allowance => {
-          setAllowance(allowance)
+          if (!cancelled) {
+            setAllowance(allowance)
+          }
         })
         .catch(err => {
           console.error(`Error fetching fee allowance : ${err}`)
         })
     }
 
     getFeeAllowance()
+
+    return () => {
+      cancelled = true
+    }
   }, [disputeManagerAddress, feeTokenContract, owner])
 
   return allowance
 }
 
-export function useTotalActiveBalancePolling(termId) {
+export function useActiveBalanceOfAt(juror, termId) {
```

Comment:
I mentioned this in https://github.com/aragon/court-dashboard/pull/179#pullrequestreview-369563941, but this seems like something the subgraph should provide rather than us having to poll for.

Is it because we could be arbitrarily asking for balances on any term, so keeping the full history in the subgraph is less ideal?

In any case, I also wonder if we should only do the polling if the requested `termId` is the current term. You would never have a previous term's active balance change, right?

Reply:
> this seems like something the subgraph should provide rather than us having to poll for.

Yes, yesterday we had a call with @facuspagnuolo about this and next week we'll have this data available in the subgraph, see https://github.com/aragon/court-subgraph/issues/47.

> In any case, I also wonder if we should only do the polling if the requested termId is the current term

Yes, note that `useActiveBalanceOfAt` does not poll but only queries.

I think that your suggestion is the way to go, `useTotalActiveBalancePolling` could be replaced by  `useTotalActiveBalance`. And we already have this implemented by @rperez89  at https://github.com/aragon/court-dashboard/pull/229/files/8c46338753c545029b88e1dd79d071817e03998b#r395011351